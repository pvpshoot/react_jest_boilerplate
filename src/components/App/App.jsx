import React, { Component } from 'react';

import './App.css';

import Tabs from '../Tabs';

class App extends Component {
  render() {
    return (
      <div>
        <h1>My React App!</h1>
        <Tabs />
      </div>
    );
  }
}

export default App;
