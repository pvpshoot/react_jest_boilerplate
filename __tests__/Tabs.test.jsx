import React from 'react';
import { mount } from 'enzyme';

import App from '../src/components/App';

describe('Tabs', () => {
  it('select tab', () => {
    const wrapper = mount(<App />);
    const tabAnchorList = wrapper.find('[data-test="tab-anchor"]');
    const tabAnchor = tabAnchorList.at(2);

    expect(wrapper.render()).toMatchSnapshot();

    tabAnchor.simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });
});
