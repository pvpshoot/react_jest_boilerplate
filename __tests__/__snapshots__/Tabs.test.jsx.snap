// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs select tab 1`] = `
<div>
  <h1>
    My React App!
  </h1>
  <button
    type="button"
  >
    Add tab
  </button>
  <div
    class="react-tabs"
    data-tabs="true"
    data-test="tab-container"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        id="react-tabs-0"
        role="tab"
      >
        Tab 1
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-anchor"
        id="react-tabs-2"
        role="tab"
        tabindex="0"
      >
        Tab 2
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        id="react-tabs-4"
        role="tab"
      >
        Tab 3
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel"
      data-test="tab-content"
      id="react-tabs-1"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      data-test="tab-content"
      id="react-tabs-3"
      role="tabpanel"
    >
      Content 2
      <button
        type="button"
      >
        remove this tab
      </button>
    </div>
    <div
      aria-labelledby="react-tabs-4"
      class="react-tabs__tab-panel"
      data-test="tab-content"
      id="react-tabs-5"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`Tabs select tab 2`] = `
<div>
  <h1>
    My React App!
  </h1>
  <button
    type="button"
  >
    Add tab
  </button>
  <div
    class="react-tabs"
    data-tabs="true"
    data-test="tab-container"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        id="react-tabs-0"
        role="tab"
      >
        Tab 1
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-anchor"
        id="react-tabs-2"
        role="tab"
        tabindex="0"
      >
        Tab 2
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        id="react-tabs-4"
        role="tab"
      >
        Tab 3
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel"
      data-test="tab-content"
      id="react-tabs-1"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      data-test="tab-content"
      id="react-tabs-3"
      role="tabpanel"
    >
      Content 2
      <button
        type="button"
      >
        remove this tab
      </button>
    </div>
    <div
      aria-labelledby="react-tabs-4"
      class="react-tabs__tab-panel"
      data-test="tab-content"
      id="react-tabs-5"
      role="tabpanel"
    />
  </div>
</div>
`;
